### ISSUE TITLE
Call of Duty: MWR - Graphic glitches during mission "blackout"

### ISSUE THREAD

**Issue by gabriele2000** [doitsujin/dxvk#3073]
### Software information
Call of Duty - Modern Warfare Remastered
Maximum settings except a few things.
I reckon the glitch goes away if I lower the quality to minimum.

### System information
- GPU: Nvidia GTX 1050TI
- Driver: 520.56.06
- Wine version: 7.20
- DXVK version: 2.0 - Release

### Apitrace file(s)
Not created, yet.

### Log files
[h1_sp64_ship_dxgi.log](https://github.com/doitsujin/dxvk/files/10024019/h1_sp64_ship_dxgi.log)
[h1_sp64_ship_d3d11.log](https://github.com/doitsujin/dxvk/files/10024020/h1_sp64_ship_d3d11.log)

[**Video**](https://www.youtube.com/watch?v=9rHWiMzegEg)

**Comment by Blisto91** [comment]
Does it also happen without the config options?

**Comment by gabriele2000** [comment]
> > info:    d3d11.samplerAnisotropy = 16
> 
> Try it without that option. It can definitely break games.
> 
> If that doesn't work, please make an apitrace.

I'll try it

**Comment by gabriele2000** [comment]
> If that doesn't work, please make an apitrace.

With DXVK the game is unbearably slow, it's even hard to see any glitch.
Without DXVK "directX error" and I cannot apitrace anything.

**Comment by K0bin** [comment]
> With DXVK the game is unbearably slow, it's even hard to see any glitch.

Doesn't matter, we need the apitrace.

**Comment by gabriele2000** [comment]
> > With DXVK the game is unbearably slow, it's even hard to see any glitch.
> 
> Doesn't matter, we need the apitrace.

Even if it's with DXVK?
Ok then, let's do it.

**Comment by gabriele2000** [comment]
[Apitrace](https://mega.nz/file/wsEgjRTL#nXegO6dtYsQJJmI_YPrXiJxtRAOAwAE00Nz5agbk5Ic)

**Comment by gabriele2000** [comment]
I'm about to try and record another apitrace, this exact type of glitches are present in other levels.
This level is faster to load, less useless frames will be captured and it's more "downloadable" for you, and easier to upload for my capped data.

**Comment by gabriele2000** [comment]
The shadows are the problem.
If I turn them off the glitches disappear.
Why?

**Comment by K0bin** [comment]
Can you post some screenshots? The apitrace looks fine.

**Comment by gabriele2000** [comment]
> Can you post some screenshots? The apitrace looks fine.
> 
> ![image](https://user-images.githubusercontent.com/1131720/202933655-6084ba7a-d0b0-4504-a08a-673310a73b90.png)

Yeah, the glitch disappear when I record the apitrace, that's just a bad joke.
Anyway, I have to try with a recent version of the game, I guess it's an issue with the game.

**Comment by Blisto91** [comment]
Adding to the issue that i can reproduce it ingame. Haven't found it to be a regression as it was the same with dxvk 1.0.
The issue appears when you have shadows turned on plus "CACHE SUN SHADOW MAPS" turned off. If you turn off shadows it seems to go away and if you have shadows on then turning on "CACHE SUN SHADOW MAPS" also seemed to make it go away.

**Comment by gabriele2000** [comment]
> turning on "CACHE SUN SHADOW MAPS" also seemed to make it go away.

Jesus, that option will just use so much VRAM that'll kill my smooth performance...

**Comment by Blisto91** [comment]
> Anyway, regression, how? 

No i meant it isn't a dxvk regression as it appeared in even the oldest dxvk versions i tried.

> What's exactly the culprit, DXVK-wise, I mean.

No idea as i have no technical knowledge about this and just try to help testing :grin: 
Also when it's buggy in this game section it has a tendency to freeze my entire graphical output so i had to force down my pc quite alot when testing this lol.

**Comment by gabriele2000** [comment]
> > Anyway, regression, how?
> 
> No i meant it isn't a dxvk regression as it appeared in even the oldest dxvk versions i tried.
> 
> > What's exactly the culprit, DXVK-wise, I mean.
> 
> No idea as i have no technical knowledge about this and just try to help testing grin Also when it's buggy in this game section it has a tendency to freeze my entire graphical output so i had to force down my pc quite alot when testing this lol.

I enabled the other cache thing too, I had to lower the shadows to "medium" because with DXVK you need to leave some room...

**Comment by K0bin** [comment]
Can you please post some screenshots? We still have no idea what the issue even looks like.

**Comment by Blisto91** [comment]
Yes i have reproduced this issue on my R9 380. Tho it is a bit annoying since it also has a tendency to hang my GPU so had to force reboot quite a few times.
I don't have any technical knowledge to begin digging deeper into it but I've tried various config options and old versions without luck so far.
Will also try to check it out on the Nvidia setup or Steam Deck.

**Comment by gabriele2000** [comment]
@Blisto91 can you record an apitrace if you have the game? Asking because, at least on my computer, the game crashes if I try to record through native wine implementation, while if I use DXVK the problem doesn't occur, since it happens at high FPS

**Comment by gabriele2000** [comment]
@K0bin I've managed to trace more glitches, while trying to replay the trace I noticed strange stuff:
- The frames with the glitches are being completely skipped
- The last frame has a gigantic number of draw-calls, like 20 frames were being combined, even more.

I'm uploading the trace now, meanwhile, could you please explain this to me?

**Comment by Blisto91** [comment]
Your trace crashes on replay but i managed to get one myself after playing at 0.4fps for half an hour. Sadly it doesn't reproduce with it. Renderdoc also makes the game crash on start.
Only clue i've found is that it seems to be a old regression in https://github.com/doitsujin/dxvk/commit/8b5db80fbd3151084e5efbf32933cae13d5e399d

**Comment by Blisto91** [comment]
Slow and painful testing (mostly because it has a tendency to hang my GPU)

**Comment by gabriele2000** [comment]
Yeah but did you inspect the trace to find it?

**Comment by Blisto91** [comment]
No i tested ingame.

**Comment by gabriele2000** [comment]
> Slow and painful testing (mostly because it has a tendency to hang my GPU)

You know what's really sad?
The only two games that gives me real problems are almost impossible to debug, can't trace them, can't view what's been traced because "yes"...

Heh, nice find by the way!
@doitsujin it is possible to try to revert that commit for testing if it solves the problem with those weird two remaster?
I hope there's a switch or something in dxvk.conf though.

**Comment by Blisto91** [comment]
> @doitsujin it is possible to try to revert that commit for testing if it solves the problem with those weird two remaster?
I hope there's a switch or something in dxvk.conf though.

Sadly not possible as we don't know what the issue is yet. The commit isn't small and touch alot of d3d11

**Comment by gabriele2000** [comment]
> Only clue i've found is that it seems to be a old regression in [8b5db80](https://github.com/doitsujin/dxvk/commit/8b5db80fbd3151084e5efbf32933cae13d5e399d)

How did you narrow the issue down to, possibly, this commit?

**Comment by Blisto91** [comment]
By gong back to very old versions and noticing it wasn't a issue there. And then afterwards bisect the specific commit it appeared

**Comment by Blisto91** [comment]
@gabriele2000 Are you able to verify that this works with dxvk master?

**Comment by gabriele2000** [comment]
> @gabriele2000 Are you able to verify that this works with dxvk master?

Once I'll reinstall the two games, I'll let you know.
It won't be long until I do, maybe less than one month.

**Comment by Blisto91** [comment]
@gabriele2000 Friendly ping

**Comment by gabriele2000** [comment]
> [@gabriele2000](https://github.com/gabriele2000) Friendly ping

I'll install the game soon, after the 24th, before the end of the month.
Reason: internet cap and I had to upload a ton of stuff those months.

**Comment by gabriele2000** [comment]
> [@gabriele2000](https://github.com/gabriele2000) Friendly ping

Everything's perfect, awesome work!
